juriscraper/pacer/email.py

```diff
@@ -293,64 +403,48 @@ def _combine_lines_with_proper_spaces(self, text):
-            return (
-                opening_tag_index,
-                closing_tag_index + 6,
-            )
-
     def _html_from_s3_email(self, text):
         """Pulls the HTML content, parsed with line breaks normalized for from the S3 email file
 
         :returns: String with proper replacements for normal HTML parsing
         """
         # Remove line ends form S3 content
         cleaned_s3_line_ends = self._combine_lines_with_proper_spaces(text)
-
-        # Find <html /> section in S3 file.
-        slice_points = self._slice_points(cleaned_s3_line_ends)
-        html_only = cleaned_s3_line_ends[
-            slice(slice_points[0], slice_points[1])
-        ]
-
-        # Clean special characters from S3.
-        replacements = [
-            (
-                r"=2E",
-                ".",
-            ),
-            (
-                r"=C2=A0",
-                " ",
-            ),
-            (
-                r"=20",
-                " ",
-            ),
-            (
-                r"=3D",
-                "=",
-            ),
-        ]
-        for replacement in replacements:
-            html_only = re.sub(replacement[0], replacement[1], html_only)
-        return html_only
+        return cleaned_s3_line_ends
 
     def _parse_text(self, text):
-        html_only = self._html_from_s3_email(text)
-        return super()._parse_text(html_only)
+        """MIME Parser from file text, text/html and text/plain messages are supported.
+        This obtains the email payload decoded as UTF-8.
+        """
+        message = email.message_from_string(text)
+        if message.is_multipart():
+            for part in message.walk():
+                c_type = part.get_content_type()
+                # If multipart message, parse text/html message
+                if c_type == "text/html":
+                    body = part.get_payload(decode=True)  # decode
+                    self.content_type = "text/html"
+                    break
+        else:
+            # If not multipart, parse either text/html or text/html message
+            for part in message.walk():
+                c_type = part.get_content_type()
+                c_dispo = str(part.get("Content-Disposition"))
+
+                if c_type == "text/html":
+                    body = part.get_payload(decode=True)
+                    self.content_type = "text/html"
+                    break
+
+                elif c_type == "text/plain" and "attachment" not in c_dispo:
+                    body = message.get_payload(decode=True)
+                    self.content_type = "text/plain"
+                    break
+
+        email_body = body.decode("UTF-8")
+        html_only = self._html_from_s3_email(email_body)
+
+        if self.content_type == "text/plain":
+            return super()._parse_text(email_body)
+        elif self.content_type == "text/html":
+            return super()._parse_text(html_only)
```

Comment:
Why not like this:

```suggestion
        if self.content_type == "text/plain":
            email_body = body.decode("UTF-8")
            return super()._parse_text(email_body)
        elif self.content_type == "text/html":
            html_only = self._html_from_s3_email(email_body)
            return super()._parse_text(html_only)
```

Reply:
Solved, I just had to leave `email_body = body.decode("UTF-8")` outside because is used in both versions